### PR TITLE
JAMES-2714 Document howTo for DeletedMessageVault

### PR DESCRIPTION
**Pull request by trantienduchn**

### CODE REVIEW

File: src/homepage/howTo/deleted-message-vault.html [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+                        <li>WebAdmin capability on top of Deleted Message Vault</li>
+                    </ul>
+
+                    <header class="major">
+                        <h2><b>What is Deleted Message Vault</b></h2>
+                    </header>
+
+                    <p>Deleted Message Vault is a plugin in James, it allows you to:</p>
+                    <ul>
+                        <li>store deleted messages from users</li>
```
Comment (chibenwa) [inline src/homepage/howTo/deleted-message-vault.html]: store user deleted messages

File: src/homepage/howTo/deleted-message-vault.html [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+                        <li>search for deleted messages by various criteria</li>
```
Comment (chibenwa) [inline src/homepage/howTo/deleted-message-vault.html]: We never search. We restore and we export.

File: src/homepage/howTo/deleted-message-vault.html [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+                        <li>delete a message by its id</li>
+                        <li>purge all expired messages, expiration time is calculated from a given configured duration called <i>retentionPeriod</i></li>
+                    </ul>
+
+                    <p>
+                        If Deleted Message Vault is enabled, when users delete their mails, along with deleting mails in the mail system,
+                        James also stores those mails into Deleted Message Vault. And only administrators can interact with this component via
```
Comment (chibenwa) [inline src/homepage/howTo/deleted-message-vault.html]: James will continue storing these mail in the deleted messages vault

File: src/homepage/howTo/deleted-message-vault.html [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+                        WebAdmin rest APIs.
+                    </p>
+                    <p>
+                        It is useful to keep users deleted messages, administrators can helps users to restore their deleted messages in case of delete by accident.
+                        But messages stored in Deleted Message Vault don't exist forever, Deleted Message Vault support to delete messages after given period had been configured
+                        (<i>retentionPeriod</i> property) by the administrators. It also supports to delete a single message by the message id.
+                    </p>
+                    <p>
+                        Deleted Message Vault API allow to search messages by various criteria base on mail properties.
```
Comment (chibenwa) [inline src/homepage/howTo/deleted-message-vault.html]: Export and restore can be filtered via several criteria:....

File: src/homepage/howTo/deleted-message-vault.html [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+                                <li>supports equal matching operator</li>
+                                <li>supports equal ignore case matching operator</li>
+                                <li>supports contains matching operator</li>
+                                <li>supports contains ignore case matching operator (with US locale)</li>
+                            </ul>
+                        </li>
+                    </ul>
+
+                    <header class="major">
+                        <h2><b>Make Deleted Message Vault available in James</b></h2>
```
Comment (chibenwa) [inline src/homepage/howTo/deleted-message-vault.html]: Please give a hands on tutorial starting from the raw dockerCompose...

 - wget xxx
 - Create configuration files
 - Add volumes for modified configuration files
 - Start James

Rationals: I should be able to try this awesome feature out without any James knowledge.

File: src/homepage/howTo/deleted-message-vault.html [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+                    </header>
+
+                    <p>
+                        To do this, you have to create a configuration file <b>deletedMessageVault.properties</b>, then put it into <b>conf</b> directory of James.
+                        There are two available properties you may want to configure:
+                    </p>
+                    <ul>
+                        <li>
+                            <b>urlPrefix</b>: represent for the prefix of namespace Deleted Message Vault will use to store deleted messages.
```
Comment (chibenwa) [inline src/homepage/howTo/deleted-message-vault.html]: For configuration details, we should avoid duplicating stuff and add links to specific configuration pages IMO

File: src/homepage/howTo/deleted-message-vault.html [hunk 7]
```diff
@@ -0,0 +1,300 @@
...
+                        retentionPeriod=1 year
+                    </code></pre>
+
+                    <p>Third, modify listener.xml to configure DeletedMessageVaultHook by adding preDeletionHook section at above paragraph</p>
+
+                    <p>Fourth, We will create a local folder for holding data out of the container:</p>
+                    <pre><code>mkdir var</code></pre>
+
+                    <pre><code>
+                        docker run
```
Comment (chibenwa) [inline src/homepage/howTo/deleted-message-vault.html]: Use docker-compose, this is much easier for newbies...

File: src/homepage/howTo/deleted-message-vault.html (hunk 2 above)
Comment (trantienduchn) [reply src/homepage/howTo/deleted-message-vault.html]: I'm talking about the DeletedMessageVault API, not the web admin

**Comment by mbaechler** [comment]
it would make sense to provide a rendered html to review, too

File: src/homepage/howTo/deleted-message-vault.html [hunk 8]
```diff
@@ -0,0 +1,300 @@
...
+        <!-- Introduction -->
+        <section id="intro" class="main special">
+            <div class="">
+                <div class="content align-left">
+                    <header class="major">
+                        <h1><b>Deleted Message Vault</b></h1>
+                    </header>
+
+                    <p>
+                        This document will help you understand what is Deleted Message Vault and what does it for. And finally,
```
[the remark this answers is not in the document]
Comment (trantienduchn) [reply src/homepage/howTo/deleted-message-vault.html]: could be more clear to replace it by `what does it do`?

File: src/homepage/howTo/deleted-message-vault.html [hunk 9]
```diff
@@ -0,0 +1,300 @@
...
+                    <p>Deleted Message Vault is a plugin in James, it allows you to:</p>
+                    <ul>
+                        <li>store deleted messages from users</li>
+                        <li>search for deleted messages by various criteria</li>
+                        <li>delete a message by its id</li>
+                        <li>purge all expired messages, expiration time is calculated from a given configured duration called <i>retentionPeriod</i></li>
+                    </ul>
+
+                    <p>
+                        If Deleted Message Vault is enabled, when users delete their mails, along with deleting mails in the mail system,
```
[the remark this answers is not in the document]
Comment (trantienduchn) [reply src/homepage/howTo/deleted-message-vault.html]: I meant `before deleting emails in James`, BTW I will delete that sentence.

**Comment by trantienduchn** [comment]
Apache James (1)

File: src/homepage/howTo/deleted-messages-vault.html [hunk 10]
```diff
@@ -0,0 +1,322 @@
...
+        ports:
+          - "80:80"
+          - "25:25"
+          - "110:110"
+          - "143:143"
+          - "465:465"
+          - "587:587"
+          - "993:993"
+          - "8000:8000"
+        volumes:
```
Comment (chibenwa) [inline src/homepage/howTo/deleted-messages-vault.html]: I don't like copying the whol docker compose here.

Can't we say:

 - dowload the docker-compose
 - Adds the following volumes for `james` service:

```
    volumes:
          - $PWD/conf:/root/conf/
          - $PWD/var:/root/var/
```

File: src/homepage/howTo/deleted-messages-vault.html [hunk 11]
```diff
@@ -0,0 +1,307 @@
...
+    $ mv keystore conf/keystore
+                    </code></pre>
+
+                    <p>Second, modify deletedMessageVault.properties configuration file</p>
+                    <pre><code>
+    urlPrefix=cassandra://var/deletedMessages/vault
+    retentionPeriod=1 year
+                    </code></pre>
+
+                    <p>Third, modify listener.xml to configure DeletedMessageVaultHook by adding preDeletionHook section at above paragraph</p>
```
Comment (Arsnael) [inline src/homepage/howTo/deleted-messages-vault.html]: s/listener.xml/listeners.xml

Comment (Arsnael) [reply src/homepage/howTo/deleted-messages-vault.html]: s/at above paragraph/as in the previous paragraph

File: src/homepage/howTo/deleted-messages-vault.html [hunk 12]
```diff
@@ -0,0 +1,307 @@
...
+                    </ul>
+
+                    <header class="major">
+                        <h3><b>Make James uses Deleted Messages Vault by configuring Pre Deletion Hooks</b></h3>
+                    </header>
+
+                    <p>
+                        By default, although Deleted Messages Vault has been configured, but, to make it really work, you still need to configure Pre Deletion Hooks to lets James use it.
+                        Before deleting a mail in James, PreDeletionHooks will be triggered to execute all hooks, if all hook executions success, then Jame will process to delete that mail.
+                        There is already a DeletedMessageVaultHook in James, it's job is storing deleted mails into Deleted Messages Vault. Thus, you need to configure this hook in listener.xml configuration file.
```
[the remark this answers is not in the document]
Comment (Arsnael) [reply src/homepage/howTo/deleted-messages-vault.html]: s/listener.xml/listeners.xml

File: src/homepage/howTo/deleted-messages-vault.html [hunk 13]
```diff
@@ -0,0 +1,307 @@
...
+                        There are two available properties you may want to configure:
+                    </p>
+                    <ul>
+                        <li>
+                            <b>urlPrefix</b>: represent for the prefix of namespace Deleted Messages Vault will use to store deleted messages.
+                        </li>
+                        <li>
+                            <b>retentionPeriod</b>: represent for the period deleted messages allowed to be stored in Deleted Messages Vault.
+                        </li>
+                    </ul>
```
Comment (Arsnael) [inline src/homepage/howTo/deleted-messages-vault.html]: I saw that there is an example a bit more down in a later section, but wouldn't it be better to put an example here instead proper to this section? (opened question)

File: src/homepage/howTo/deleted-messages-vault.html [hunk 14]
```diff
@@ -208,31 +208,16 @@ <h3><b>Starting James with enabled Deleted Message Vault by docker compose</b></
...
-      james_elastic:
-        image: elasticsearch:2.4.6
-      cassandra:
-        image: cassandra:3.11.3
-      tika:
-        image: logicalspark/docker-tikaserver:1.20
+    $ wget https://raw.githubusercontent.com/apache/james-project/master/dockerfiles/run/docker-compose.yml
+                    </code></pre>
+
+                    <p>Adds the following volumes for james service:</p>
```
Comment (rouazana) [inline src/homepage/howTo/deleted-messages-vault.html]: s/Adds/Add

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**